types/evidence.go

```diff
@@ -644,10 +645,26 @@ OUTER_LOOP:
 				} else {
 					// if H1.Round != H2.Round we need to run full detection procedure => not
 					// immediately slashable.
-					evList = append(evList, &PotentialAmnesiaEvidence{
-						VoteA: ev.H1.Commit.GetVote(int32(i)),
-						VoteB: ev.H2.Commit.GetVote(int32(j)),
-					})
+					firstVote := ev.H1.Commit.GetVote(int32(i))
+					secondVote := ev.H2.Commit.GetVote(int32(j))
+					var newEv PotentialAmnesiaEvidence
+					if firstVote.Timestamp.Before(secondVote.Timestamp) {
```

Comment:
shouldn't we just enforce order in ValidateBasic? if we should (or we already do this), why have this if else condition here?

Reply:
Yes but we need to make sure we generate them in the right order